Code/Entities/ClimbableOneWayPlatform.cs

```diff
@@ -130,21 +156,32 @@ public class ClimbableOneWayPlatform : Entity {
             }
         }
 
-        private static bool checkCollisionWithSidewaysMovingPlatformsWhileMoving(Actor self, int moveDirection, bool movingLeftToRight) {
+        private static bool checkCollisionWithSidewaysMovingPlatformsWhileMoving(Actor self, int moveDirection, bool movingLeftToRight)
+        {
             ClimbableOneWayPlatform climbablePlatform = collideFirstOutside(self, self.Position + Vector2.UnitX * moveDirection, !movingLeftToRight);
-            return Input.Grab.Check && self is Player player && player.Stamina >= 20f && climbablePlatform != null && climbablePlatform.climbJumpGrabCooldown <= 0f;
+            if (climbablePlatform != null && climbablePlatform.stamBehavior == "none")
+            {
+                return Input.Grab.Check && self is Player player && player.Stamina >= 20f && climbablePlatform != null && climbablePlatform.climbJumpGrabCooldown <= 0f;
+            }
+
+            //if a custom stamina behavior is present, set the stamina to slightly above red point
+            if (Input.Grab.Check && self is Player && climbablePlatform != null && climbablePlatform.climbJumpGrabCooldown <= 0f)
+                (self as Player).Stamina = Math.Max((self as Player).Stamina, 21f);
+            return Input.Grab.Check && self is Player && climbablePlatform != null && climbablePlatform.climbJumpGrabCooldown <= 0f;

```

Comment:
you could put `Input.Grab.Check && self is Player && climbablePlatform != null && climbablePlatform.climbJumpGrabCooldown <= 0f` in a bool variable then re-use it 3 times in this method

Reply:
Applied change